src/test/ui/macros/issue-88206.stderr

```diff
@@ -32,15 +32,15 @@ error: cannot find macro `Box` in this scope
 LL |     Box!();
    |     ^^^
    |
-   = note: `Box` is in scope, but it is a struct
+   = note: `Box` is in scope, but it is a struct, not a macro
```

Comment:
For the cases where we don't have a span, wouldn't it make sense to turn this into the primary span? Something like:

```
error: cannot find macro `Copy` in this scope
  --> $DIR/issue-88206.rs:56:5
   |
LL |     Copy!();
   |     ^^^^ this is not a macro, it is a derive macro: `#[derive(Copy)]`
```

Reply:
I'd avoid pointing the note at the identifier the user wrote, because the error already says `` cannot find `Thing` ``, and the note is about *another* `Thing`, not the `Thing` they wrote. (Even though they have the same name.)

So in the case we can point at an import, we do that to highlight `Thing` here is not found, but `Thing` there exists but is something else. Pointing both at the same identifier would make it less clear that we're talking about two different `Thing`s.